### ISSUE TITLE
Very slow when operating with Commit dialog's list

### ISSUE THREAD

**Issue by GoogleCodeExporter** [guberm/tortoisegit#1196]
```
What steps will reproduce the problem?
1. Open Commit dialog having many files to add (>10'000)
2. Wait some minutes to see there are some unversioned files
3. Right-click on single unversioned file
4. Click "Add"
5. You'll have to wait some minutes again just everything to be refreshed.

What is the expected output? What do you see instead?
After step 4 not to wait for refreshing of whole files list. But the 
unversioned file I tried to add to change in-place in the list without 
refreshing whole list.

What version of the product are you using? On what operating system?
1.7.9 on Windows 7 x64

Please provide any additional information below.

```

Original issue reported on code.google.com by `var...@gmail.com` on 22 May 2012 at 3:22
- Merged into: #1122
